.github/workflows/cli-tests.yaml

```diff
@@ -36,14 +31,27 @@ jobs:
     timeout-minutes: 10
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v3.5.0
         with:
           go-version-file: ./go.mod
-          cache: false # use golangci cache instead
+      - name: Get current time
+        uses: josStorer/get-current-time@v2.0.2
+        id: current-time
+      - name: Mount golangci-lint cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/golangci-lint
+            ~/.cache/go-build
+            ~/go/pkg
+          key: golangci-lint-cache-${{ runner.os }}-${{ steps.current-time.outputs.day }}
```

Comment:
Should the key be the hash of `go.mod` file?

Reply:
Caching with go.mod key creates new keys very frequently (without much benefit). Currently (after weekend) we're using 12GB of 10GB max (so pretty much all new data causes eviction). On Friday we were using 30GB, which means we're cycling though keys every 8 hours or so. 

This is a tradeoff. It means the cache is a little less useful, (but in practice pretty good), and we never get cache misses and also use a lot less storage. If GHA had unlimited storage, I would run jobs on `main` that cache with both keys, but since storage is pretty limited this feels like the right approach.